x/ecocredit/client/tx.go

```diff
@@ -79,40 +79,85 @@ Parameters:
 	}
 }
 
+const (
+	FlagIssuer          string = "issuer"
+	FlagClassId         string = "class-id"
+	FlagStartDate       string = "start-date"
+	FlagEndDate         string = "end-date"
+	FlagProjectLocation string = "project-location"
+	FlagMetadata        string = "metadata"
+	FlagIssuance        string = "issuance"
+)
+
 func txCreateBatch() *cobra.Command {
-	return &cobra.Command{
-		Use:   "create-batch [issuer] [class_id] [start_date] [end_date] [metadata] [issuance]",
+	cmd := &cobra.Command{
+		Use:   "create-batch [--issuer issuer] [--class-id class_id] [--start-date start_date] [--end-date end_date] [--project-location project_location] [--metadata metadata] [--issuance issuance]",
```

Comment:
I  think having a JSON parameter is better than all this command parameters, eg:
```
create-batch msg-create-batch
Parameters:
   msg-create-batch   a JSON object representing MsgCreateBatchRequest: %s  ...", json.Marshal(MsgCreateBatchRequest{})
```

With this, parsing we will simplify parameters parsing.

Reply:
I had the same thought but then it's just basically the whole MsgIssueBatch json which is maybe okay

Reply:
So, I agree this would probably be better for UX. I think it would be helpful to add some sort of util that give people a template JSON file or even lets them pass some of the values as command line params. Then they can manually inspect the JSON and then run:
```
regen tx ecocredit create-batch < batch.json
```

Reply:
The util could take:
- Class ID string
- Number of issuances
- Metadata as a string
- Start date in yyyy-mm-dd format
- End date in yyyy-mm-dd format
- Project location string

This then creates a JSON file which has:
- Template issuances the user to fill in
- Base64 encoded the metadata
- Put the start date and end date into the full timestamp format for parsing

This basically takes care of all the tedious/tricky bits of getting the JSON file right.

Thoughts?

Reply:
I think that makes sense. I don't think we need to spend a ton of time on it yet, but if it's simple sure. I'm sort of of the opinion the metadata should just be a binary hash and nothing more at this point. But maybe a string is fine. Also we would keep this under the `tx` command set

Reply:
@aaronc It was fairly quick to implement, so I've done it. Would you give it a quick look over?